arm9/source/main.c

```diff
@@ -108,18 +109,56 @@ void dump_firmware(u8* buffer, u32 size) {
 	fifoGetValue32(FIFO_RETURN);
 }
 
-bool write_file(char* path, u8* buffer, u32 size) {
-	printf("Saving %s\n", path+16);
-	FILE* f = fopen(path, "wb");
-	if(!f) {
-		printf("Failed to open %s\n", path);
-		return false;
+bool save_dump(char* path, u8* buffer, u32 size) {
+	if (!isNitroUnit) {
+		printf("Saving %s\n", path+16);
+		FILE* f = fopen(path, "wb");
+		if(!f) {
+			printf("Failed to open %s\n", path);
+			return false;
+		}
+		fseek(f, 0, SEEK_SET);
+		fwrite(buffer, size, 1, f);
+		fflush(f);
+		fclose(f);
+		return true;
+	} else {
+		// If we're on a nitro unit, write the values into memory so they can be
+		// fetched from the host machine, through is-nitro-debugger or a custom
+		// tool.
+		//
+		// We fill the area around the buffer with very recognizable values, so
+		// they can be very easily searched for.
+		u8* data = malloc(size + 512);
+		memset(data, 0xDE, 256);
+		memset(data+256+size, 0xAD, 256);
+		memcpy(data+256, buffer, size);
+		printf("buffer at: %p\n", data);
+		printf("If manually dumping press START to continue when you're ready.\n");
+		printf("If using a tool we will automatically continue when we saved the dump.\n");
```

Comment:
Could you add newlines where the text might not fit on the screen? This line is quite long and will certainly wrap around the console, it might cut off somewhere.

(I'd appreciate if you would do it for all the new printfs even if they do wrap without cutting off.)

Reply:
Yeah, not a problem at all, happy to fix it up!